src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java

```diff
@@ -455,18 +467,35 @@ public ObTableBatchOperationResult executeInternal() throws Exception {
         if (tableName == null || tableName.isEmpty()) {
             throw new IllegalArgumentException("table name is null");
         }
-        long start = System.currentTimeMillis();
         List<ObTableOperation> operations = batchOperation.getTableOperations();
-        final ObTableOperationResult[] obTableOperationResults = new ObTableOperationResult[operations
-            .size()];
+        if (operations.isEmpty()) {
+            throw new IllegalArgumentException("operations is empty");
+        }
+        ObTableOperationType lastType = operations.get(0).getOperationType();
+        if (returnOneResult
+                && !(batchOperation.isSameType() && (lastType == ObTableOperationType.INSERT
```

Comment:
can we move this defensive code to the interface layer, which will be more readable?

Reply:
done